daemon-scheduler/pkg/engine/monitor_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package engine
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	mocks "github.com/blox/blox/daemon-scheduler/pkg/mocks"
+	"github.com/blox/blox/daemon-scheduler/pkg/types"
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type MonitorTestSuite struct {
```

Comment:
The tests actually do hit ctx.Done() because that's how you break out of the for loop. If they didn't, the for loop would never break and the tests would not finish